classes/CartRule.php

```diff
@@ -1011,6 +1018,16 @@ public function checkValidity(Context $context, $alreadyInCart = false, $display
             return (!$display_error) ? false : $this->trans('You cannot use this voucher because it has manually been removed.', [], 'Shop.Notifications.Error');
         }
 
+        // Check if another discount is already applied
```

Comment:
Since there are other comments to handle we might as well take the opportunity to improve the comments to reassure people

Reply:
```suggestion
        // This part introduces the new business rules for the discount rework they are only taking effect when the discount feature flag is enabled
```